rules/react.js

```diff
@@ -1,4 +1,540 @@
+    'react/display-name': ['off', { ignoreTranspilerName: false }],
+
+    // Enforce boolean attributes notation in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md
+    'react/jsx-boolean-value': ['error', 'never', { always: [] }],
+
+    // Validate closing bracket location in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-bracket-location.md
+    'react/jsx-closing-bracket-location': ['error', 'line-aligned'],
+
+    // Validate closing tag location in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-tag-location.md
+    'react/jsx-closing-tag-location': ['error'],
+
+    // Enforce or disallow spaces inside of curly braces in JSX attributes
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-curly-spacing.md
+    'react/jsx-curly-spacing': ['error', 'never', { allowMultiline: true }],
+
+    // Enforce event handler naming conventions in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md
+    'react/jsx-handler-names': [
+      'error',
+      {
+        eventHandlerPrefix: 'handle',
+        eventHandlerPropPrefix: 'on',
+      },
+    ],
+
+    // Validate props indentation in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-indent-props.md
+    'react/jsx-indent-props': ['error', 2],
+
+    // Validate JSX has key prop when in array or iterator
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-key.md
+    'react/jsx-key': 'error',
+
+    // Limit maximum of props on a single line in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-max-props-per-line.md
+    'react/jsx-max-props-per-line': [
+      'error',
+      { maximum: 1, when: 'multiline' },
+    ],
+
+    // Prevent usage of .bind() in JSX props
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md
+    'react/jsx-no-bind': [
+      'error',
+      {
+        ignoreRefs: true,
+        allowArrowFunctions: true,
+        allowFunctions: false,
+        allowBind: false,
+        ignoreDOMComponents: true,
+      },
+    ],
+
+    // Prevent duplicate props in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-duplicate-props.md
+    'react/jsx-no-duplicate-props': ['error', { ignoreCase: true }],
+
+    // Disallow undeclared variables in JSX
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-undef.md
+    'react/jsx-no-undef': 'error',
+
+    // Enforce PascalCase for user-defined JSX components
+    // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-pascal-case.md
+    'react/jsx-pascal-case': ['error', { allowAllCaps: true }],
+
+    'react/jsx-sort-props': [
+      'error',
```

Comment:
@taigakiyokawa 
Applying clear alphabetical rules brings consistency to your code and definitely improves the readability of your code.

> Some props in a product may have been sorted depending on their functions, relationships with other components, or just readable or maintainable than alphabetically by the developer team.

Such development teams can override this rule. This rule is set for teams that do not have their own hopes.

Reply:
@wakamsha I don't think sorting props alphabetically **definitely** improves the readability. By the way, I misunderstood the options. You've set `noSortAlphabetically: true` which doesn’t sort props alphabetically. So, it seems to be OK to enable it because it's not as strict as I'm concerned 🙆‍♂️